packages/cli/src/scaling/scaling.service.ts

```diff
@@ -41,6 +42,8 @@ import type {
 export class ScalingService {
 	private queue: JobQueue;
 
+	private queueReadyPromise = createDeferredPromise();
```

Comment:
Maybe we could replace `private queue: JobQueue` with `private queuePromise: Promise<JobQueue>` and use that everywhere so we don't have to keep the queue and its promise separately?